### DISCUSSION TITLE
描边绘制有问题

### DISCUSSION THREAD

**Discussion by shield2018** [antvis/G2#4984] [Q&A] [answered]
v5.0.1

柱状图设置了stroke，绘制出来上下有很小的偏移，导致上部元素颜色和边框颜色重叠，会有一条深色线，底部会有一条白线
macbook高分屏幕，放大后截图
![big](https://user-images.githubusercontent.com/5371329/230356681-a8ad8954-510c-45fb-9ed1-cb03735389b8.png)
macbook高分屏幕，未放大
![small](https://user-images.githubusercontent.com/5371329/230356314-e65e2459-aa4b-4388-a8ea-d1713547cac0.png)

**Comment by xiaoiver** [comment]
我写了一个 demo，用原生 Canvas API 绘制的：
https://codepen.io/xiaoiver/pen/KKGMMqB

<img width="254" alt="截屏2023-04-17 下午4 13 30" src="https://user-images.githubusercontent.com/3608471/232425448-86f4bcb7-b6a4-49ac-a702-1b3fcb492593.png">

你可以看看你那边是否也有类似现象？

**Comment by shield2018** [comment]
> 

demo是很多小球闪动？

**Comment by xiaoiver** [comment, marked as answer]
> demo是很多小球闪动？

不好意思之前的 demo 被覆盖了，我又写了一个：
https://codepen.io/xiaoiver/pen/dygmNjE

我看了下目前的实现也就是调用原生的 Canvas2D API 例如 `ctx.fill / stroke`。

**Reply by shield2018** [reply]
![A635A62F-2526-4B15-8892-B43B97C65FAC](https://github.com/antvis/G2/assets/5371329/b6e0bd7d-9cb8-4c2e-9a1a-b84d8f411a7c)
确实也是这样

**Reply by hustcc** [reply]
所以这个问题可能是普遍都存在的问题嘛？
